### PR TITLE
Update dependencies to latest, and other fixes

### PR DESCRIPTION
**Pull request by a-maurice**
Update to the latest C++ SDK, and the equivalent Android/iOS dependencies.  Update the versions of python and Node used by the GHA runners, because the CLI needs it.  Increase the timeout range on the macOS build, since the build seems to be taking longer because of dependency changes.
***

Runs on another branch with these changes
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [x] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: docs/readme.md [hunk 1]
```diff
@@ -71,6 +71,12 @@ Support
 
 Release Notes
 -------------
+### Upcoming
```
Comment (cynthiajoan) [inline docs/readme.md:74]: still upcoming?

**Review by cynthiajoan — APPROVED** [review]

File: docs/readme.md (hunk 1 above)
Comment (a-maurice) [reply docs/readme.md:74]: Yeah, not updating the version number of the Unity SDK yet, so upcoming (though hopefully not too much longer)

Comment (rubenwe) [reply docs/readme.md:74]: Looking forward to it! We really, really need the updated versions. Thanks!

Comment (a-maurice) [reply docs/readme.md:74]: Sorry for the delays, we are still fixing a problem with the Android dependencies, which we are still testing and iterating against different Unity editor versions, but again, hopefully will be available soon.

**Pull request merged**